src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -240,6 +240,33 @@ install_packages (RpmOstreeTreeComposeContext  *self,
     if (!rpmostree_context_setup (self->corectx, tmprootfs_abspath, NULL, self->treespec,
                                   cancellable, error))
       return FALSE;
+
+#define TMP_SELINUX_ROOTFS "selinux.tmp/etc/selinux"
+
+    /* By default, the core starts with the SELinux policy of the root, but if we have a
+     * previous commit, it's much likelier that its policy will be closer to the final
+     * policy than the host system's policy. And in the case they match, we skip a full
+     * relabeling phase. Let's use that instead. */
+    if (self->previous_checksum)
+      {
+        if (!glnx_shutil_mkdir_p_at (self->workdir_dfd,
+                                     dirname (strdupa (TMP_SELINUX_ROOTFS)), 0755,
+                                     cancellable, error))
+          return FALSE;
+        OstreeRepoCheckoutAtOptions opts = { .subpath = "/usr/etc/selinux" };
+        if (!ostree_repo_checkout_at (self->repo, &opts, self->workdir_dfd,
+                                      TMP_SELINUX_ROOTFS, self->previous_checksum,
+                                      cancellable, error))
+          return FALSE;
+
+        g_autofree char *abspath = glnx_fdrel_abspath (self->workdir_dfd, "selinux.tmp");
+        g_autoptr(GFile) path = g_file_new_for_path (abspath);
```

Comment:
We can use `ostree_sepolicy_new_at()` here right?

Reply:
Yeah, that was my initial patch, though I switched it to using `GFile` to avoid having to own yet another dfd. Open to changing it if you'd prefer.